[feat] 관심상품 등록/해제 기능

pie2457 opened the pull request:
## Issues
- #30 

## What is this PR? 👓
- 관심상품 등록 / 해제 기능에 관한 PR입니다.

## Key changes 🔑
- 관심 상품 등록/해제 기능
- 관심 상품 등록/해제 테스트 코드

pie2457 committed:
feat #30 : 관심상품 등록/해제 기능

pie2457 committed:
feat #30 : 에러코드 작성

pie2457 committed:
test #30 : 테스트코드 작성

23Yong commented on src/main/java/pie/tomato/tomatomarket/application/wish/WishItemService.java:
```diff
@@ -0,0 +1,51 @@
+
+	private void cancel(Item item) {
+		item.wishCancel();
+		wishRepository.deleteByItemId(item.getId());
```
이렇게 되면 `itemId`를 가진 모든 관심 등록 목록이 사라질 것 같아요~
```java
wishRepository.deleteByItemIdAndMemberId(item.getId(), member.getId());
```

특정 멤버가 등록한 관심상품 등록을 해제해야 할 것 같네요~!

pie2457 replied:
헉 그렇네요 :O 감사합니다 수정할게요~!

23Yong commented on src/main/java/pie/tomato/tomatomarket/domain/WishStatus.java:
```diff
@@ -0,0 +1,21 @@
+
+	public static WishStatus of(String status) {
+		return Arrays.stream(WishStatus.values())
+			.filter(wishStatus -> wishStatus.name().equals(status.toUpperCase()))
```
`wishStatus.name().equalsIgnoreCase(status)`도 좋을 것 같습니다~ 그냥 고려만 해봐주세용

pie2457 replied:
넵 감사합니당 !

23Yong commented on src/test/java/pie/tomato/tomatomarket/application/WishItemServiceTest.java:
```diff
@@ -0,0 +1,89 @@
+			"역삼1동", 0L, 0L, 0L, LocalDateTime.now(), member, category));
+
+		// when
+		wishItemService.changeWishStatus(item.getId(), "YES", principal);
+		wishItemService.changeWishStatus(item.getId(), "YES", principal);
+		wishItemService.changeWishStatus(item.getId(), "YES", principal);
```
같은 회원이 같은 상품에 대해 관심 등록을 세 번하면 `wishCount`가 3개 올라가네요~ 의도하신건가요~?

pie2457 replied:
넹 일단 올라가는지 테스트 해보고 나중에 프론트를 이어붙인다면 프론트 쪽에서 관심이 눌러져있는 상태에서 또 누른다면 취소되게끔 할 생각입니다 !

23Yong reviewed (approved):
수고하셨습니다~!

pie2457 committed:
refactor #30 : 리뷰사항 반영

- WishItemService : cancel 메서드에 member를 같이 넘겨 member가 누른 item에 대해서만 관심상품 해제하도록 수정
- WishRepository : memberId에 해당하는 관심상품만 해제되도록 memberId를 같이 받게 수정
